fs/api_internal.go

```diff
@@ -1289,7 +1305,7 @@ func (mS *mountStruct) MiddlewareGetAccount(maxEntries uint64, marker string) (a
 	lastBasename := marker
 	for areMoreEntries && uint64(len(accountEnts)) < maxEntries {
 		var dirEnts []inode.DirEntry
-		dirEnts, _, areMoreEntries, err = mS.Readdir(inode.InodeRootUserID, inode.InodeRootGroupID, nil, inode.RootDirInodeNumber, lastBasename, maxEntries-uint64(len(accountEnts)), 0)
+		dirEnts, _, areMoreEntries, err = mS.Readdir(inode.InodeRootUserID, inode.InodeGroupID(0), nil, inode.RootDirInodeNumber, lastBasename, maxEntries-uint64(len(accountEnts)), 0)
```

Comment:
I do not agree with the change to convert inode.InodeRootGroupID to inode.InodeGroupID(0)...
There is, in fact, a "root" GroupID being zero on both CentOS and Ubuntu from what I can see and do not think we need to "avoid" it in the calls.

That said, as far as I understand it, the fact that inode.InodeRootUserID is passed obviates the need to even ever consult the gid argument :-).

Reply:
thanks for your reply.  i'm sorry i didn't get to it sooner!

the problem that i see with "InodeRootGroupID" is that people seem to assume it has special priviledges, probably because InodeRootUserID does (see the code in Access() that i deleted), but there are no special privileges associated with Group ID 0 so i don't think it deserves a special name.  for example, we don't have a InodeBinGroupID (gid 2 on Ubuntu and Centos and probably everywhere else).

so i wanted to demote it to "place holder" status.

what if we rename it to InodeGroupDummyID?

Reply:
it looks like MiddlewareCoalesce() creates a directory uses InodeRootUserID and InodeRootGrooupID (UID 0 and GID 0) and permissions 0755.

however, putObjectHelper() (for MiddlewarePutComplete()) just uses 0 for the UID and GID and 0777 for permissions on the directory(s) and file.  it also bypasses permission checking on its way to the file, so i think MiddlewarePut() can overwrite any file in the file system.  the new objects probably shouldn't be executable!